tests/test_fixtures.py

```diff
@@ -1,2 +1,131 @@
 # check info function of returned structs of each fixture
 # test correct working directory
+import os
+import pathlib
+import tempfile
+
+import pytest
+
+from .fixtures import to_memoized_dir, to_tmp_dir  # noqa
+
+
+def test_to_tmp_dir():
+    cwd = os.getcwd()
+    with to_tmp_dir() as tmp_dir:
+        print(tmp_dir, cwd, os.getcwd())
+
+        assert str(tmp_dir) == os.getcwd()
+        assert cwd != os.getcwd()
+        assert len(os.listdir(tmp_dir)) == 0
+
+    assert cwd == os.getcwd()
+    assert not os.path.isdir(str(tmp_dir))
+
+
+def test_fix_tmp_dir(tmp_dir):
+    assert len(os.listdir(tmp_dir)) == 0
+
+
+def test_fix_tmp_dir_session():
```

Comment:
Aha ... that's what I get for reading from the top ... hmm... not sure you need this isolation though... lets talk about it.